executor/point_get.go

```diff
@@ -105,15 +106,24 @@ func (e *PointGetExecutor) Next(ctx context.Context, req *chunk.Chunk) error {
 		return nil
 	}
 	e.done = true
+	txnCtx := e.ctx.GetSessionVars().TxnCtx
 	snapshotTS := e.startTS
 	if e.lock {
-		snapshotTS = e.ctx.GetSessionVars().TxnCtx.GetForUpdateTS()
+		snapshotTS = txnCtx.GetForUpdateTS()
 	}
 	var err error
-	e.snapshot, err = e.ctx.GetStore().GetSnapshot(kv.Version{Ver: snapshotTS})
+	e.txn, err = e.ctx.Txn(false)
```

Comment:
`c.ctx.Txn(true)` ?

Reply:
Sometimes we don't need to create a transaction for point get.

Reply:
It doesn't block using `c.ctx.Txn(true)` here, because the starts have already been obtained here https://github.com/pingcap/tidb/blob/1b3b6f0e80b462e32182ef0cbb8ba66d2fb41604/executor/adapter.go#L199

Well, it just looks more natural for the code, if you insist, I'm fine with it.